airflow/api_fastapi/core_api/routes/public/tasks.py

```diff
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from fastapi import HTTPException, Request, status
+
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.api_fastapi.core_api.serializers.tasks import TaskResponse
+from airflow.exceptions import TaskNotFound
+from airflow.models import DAG
+
+tasks_router = AirflowRouter(tags=["Task"], prefix="/dags")
+
+
+@tasks_router.get(
+    "/{dag_id}/tasks/{task_id}",
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_400_BAD_REQUEST,
+            status.HTTP_401_UNAUTHORIZED,
+            status.HTTP_403_FORBIDDEN,
+            status.HTTP_404_NOT_FOUND,
+        ]
+    ),
+)
+async def get_task(dag_id: str, task_id, request: Request) -> TaskResponse:
+    """Get simplified representation of a task."""
+    dag: DAG = request.app.state.dag_bag.get_dag(dag_id)
+    if not dag:
+        raise HTTPException(status.HTTP_404_NOT_FOUND, f"Dag with id {dag_id} was not found")
+    try:
+        task = dag.get_task(task_id=task_id)
```

Comment:
Calling aync method from async function?

Reply:
FastAPI knows how to run sync code from default async routes. For now all the API code is sync because we are missing some async utility code in airflow core.

This will be changed in the future when we have time to tackle that.

Reply:
> FastAPI knows how to run sync code from default async routes

That's not true. There is no special magic under the hood of FastAPI. It just wraps all sync endpoints (`def`) into `asyncio.to_thread`, sending them to ThreadPool, and not blocking the current event loop. For `async def` endpoints, it does not do that.

Reply:
True, I got confused, I thought It was run in an external threadpool, but that's only  true for `non async` route.I think we need to remove all of the `async` keyword as long as the code has blocking sync operations.

I'll open a PR for that, thanks for pointing it.

Reply:
PR is ready here, https://github.com/apache/airflow/pull/43797

Thanks again.

Reply:
> Calling sync method (making network/IO requests) from async function? Doesn't this block the event loop?

Thanks for highlighting this @dolfinus! Yes the sync func will block the event loop for that particular thread. We can probably explore and consider wrapping the sync funcs in async and then call those from the async path funcs. Please see [this comment](https://github.com/apache/airflow/pull/43797#issuecomment-2463876083).

Reply:
ok @pierrejeambrun i have added an asyncio "hello world" for airflow

maybe then we can actually implement async as we migrate????

https://github.com/apache/airflow/pull/43944